app/uk/gov/hmrc/slacknotifications/services/NotificationService.scala

```diff
@@ -86,34 +94,35 @@ class NotificationService @Inject()(
-          sendSlackMessage(
-            SlackMessage(
-              channel     = slackConfig.noTeamFoundAlert.channel,
-              text        = slackConfig.noTeamFoundAlert.text.replace("{service}", service.name),
-              username    = slackConfig.noTeamFoundAlert.username,
-              icon_emoji  = Some(slackConfig.noTeamFoundAlert.iconEmoji),
-              attachments = notificationRequest.messageDetails.attachments,
-              showAttachmentAuthor = true
-            ),
-            service     = service
-          )
-          Future.successful(NotificationResult().addError(TeamsNotFoundForUsername(userType, username)))
-        }
-      }
-    }
-  }
+  ): Future[NotificationResult] =
+  for {
+    allTeams                  <- teamsGetter(username)
+    (excluded, toBeProcessed)  = allTeams.map(_.team).partition(slackNotificationConfig.notRealTeams.contains)
+    notificationResult        <- if (toBeProcessed.nonEmpty) {
+                                  toBeProcessed.foldLeftM(Seq.empty[NotificationResult]) { (acc, teamName) =>
+                                    for {
+                                      slackChannel <- withExistingSlackChannel(teamName)
+                                      notificationRes <- sendSlackMessage(fromNotification(notificationRequest, slackChannel), service)
+                                    } yield acc :+ notificationRes
+                                  }}.map(flatten)
+                                else {
+                                    logger.info(s"Failed to find teams for usertype: ${userType}, username: ${username}. " +
+                                      s"Sending slack notification to Platops admin channel instead")
+                                    sendSlackMessage(
+                                      SlackMessage(
+                                        channel = slackConfig.noTeamFoundAlert.channel,
+                                        text = slackConfig.noTeamFoundAlert.text.replace("{service}", service.name),
+                                        username = slackConfig.noTeamFoundAlert.username,
+                                        icon_emoji = Some(slackConfig.noTeamFoundAlert.iconEmoji),
+                                        attachments = notificationRequest.messageDetails.attachments,
+                                        showAttachmentAuthor = true
+                                      ),
+                                      service = service
+                                    )
+                                    Future.successful(NotificationResult().addError(TeamsNotFoundForUsername(userType, username)))
+                                }
+    resWithExclusions          = notificationResult.addExclusion(excluded.map(NotARealTeam.apply): _*)
+  } yield resWithExclusions
 
   private def flatten(results: Seq[NotificationResult]): NotificationResult =
```

Comment:
can we renamed this, the name is confusing - it' not flattening nested structure, but concatenating a collection of the same type into one instance  - `concat` might be a better name.

Reply:
Changed to `concatResults` (thought maybe `concat` still clashes a little bit with the built-in method)